openhands/resolver/send_pull_request.py

```diff
@@ -426,10 +426,13 @@ def update_existing_pull_request(
 
     # Reply to each unresolved comment thread
     if additional_message and github_issue.thread_ids:
-        explanations = json.loads(additional_message)
-        for count, reply_comment in enumerate(explanations):
-            comment_id = github_issue.thread_ids[count]
-            reply_to_comment(github_token, comment_id, reply_comment)
+        try:
+            explanations = json.loads(additional_message)
+            for count, reply_comment in enumerate(explanations):
+                comment_id = github_issue.thread_ids[count]
+                reply_to_comment(github_token, comment_id, reply_comment)
+        except (json.JSONDecodeError, TypeError):
```

Comment:
I wonder if we can make this a little more robust and user-friendly: if `json.loads()` fails, fall back to loading it as a string and still send it in a comment. It'd give us easier visibility into what happened, rather than crawling through workflow logs.

Reply:
That's a good point! We leave a regular comment with a summary of all threads replies if the deserialization fails.